File: source/XYO/QuantumScript/LibStdError.js
// Quantum Script
// Copyright (c) 2016-2023 Grigore Stefan <g_stefan@yahoo.com>
// MIT License (MIT) <http://opensource.org/licenses/MIT>
// SPDX-FileCopyrightText: 2016-2023 Grigore Stefan <g_stefan@yahoo.com>
// SPDX-License-Identifier: MIT

function Error(message) {
	this.message = message;
};

Error.prototype.toString = function() {
	return "Error: " + this.message;
};

File: source/XYO/QuantumScript/LibStdError.cpp
// Quantum Script
// Copyright (c) 2016-2025 Grigore Stefan <g_stefan@yahoo.com>
// MIT License (MIT) <http://opensource.org/licenses/MIT>
// SPDX-FileCopyrightText: 2016-2025 Grigore Stefan <g_stefan@yahoo.com>
// SPDX-License-Identifier: MIT

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#include <XYO/QuantumScript/LibStdError.hpp>

#include <XYO/QuantumScript/LibStdError.Source.cpp>

// #define XYO_QUANTUMSCRIPT_DEBUG_RUNTIME

namespace XYO::QuantumScript {

	namespace LibStdError {

		void initExecutive(Executive *executive, void *extensionId) {
			executive->compileStringX(libStdErrorSource);
		};

	};
};
